Library/ENV/scm/git

```diff
@@ -1,71 +1,113 @@
-#!/bin/sh
-# Make sure this shim uses the same Ruby interpreter that is used by Homebrew.
-unset RUBYLIB
-unset RUBYOPT
-if [ -z "$HOMEBREW_RUBY_PATH" ]
-then
-  echo "${0##*/}: The build tool has reset ENV; --env=std required." >&2
-  exit 1
-fi
-exec "$HOMEBREW_RUBY_PATH" -x "$0" "$@"
-#!/usr/bin/env ruby -W0
+#!/bin/bash
 
 # This script because we support $GIT, $HOMEBREW_SVN, etc., Xcode-only and
 # no Xcode/CLT configurations. Order is careful to be what the user would want.
 
-require "pathname"
+set +o posix
+
+quiet_cd() {
+  cd "$@" >/dev/null
+}
+
+realpath() {
+  local path="$1"
+  local dir
+  local base
+
+  dir="$(quiet_cd "${path%/*}/" && pwd -P)"
+  base="${path##*/}"
+  path="$dir/$base"
 
-SELF_REAL = Pathname.new(__FILE__).realpath
-F = File.basename(__FILE__).freeze
-D = File.expand_path(File.dirname(__FILE__)).freeze
+  while [[ -L "$path" ]]
+  do
+    path="$dir/$(readlink "$path")"
+    dir="$(quiet_cd "${path%/*}/" && pwd -P)"
+    base="${path##*/}"
+    path="$dir/$base"
+  done
 
-def executable?(file)
-  File.file?(file) && File.executable?(file)
-end
+  echo "$path"
+}
 
-def exec(*args)
+executable() {
+  local file="$1"
+  [[ -f "$file" && -x "$file" ]]
+}
+
+lowercase() {
+  echo "$1" | tr '[:upper:]' '[:lower:]'
+}
+
+safe_exec() {
+  local arg0="$1"
+  if ! executable "$arg0"
+  then
+    return
+  fi
   # prevent fork-bombs
-  arg0 = args.first
-  return if arg0 =~ /^#{F}/i || Pathname.new(arg0).realpath == SELF_REAL
-  if ARGV == %w[--homebrew=print-path]
-    puts arg0
+  if [[ "$(lowercase "$arg0")" = "$F" || "$(realpath "$arg0")" = "$SELF_REAL" ]]
+  then
+    return
+  fi
+  if [[ "$HOMEBREW" = "print-path" ]]
+  then
+    echo "$arg0"
     exit
-  end
-  super
-end
-
-case F.downcase
-when "git" then %W[HOMEBREW_GIT GIT]
-when "svn" then %W[HOMEBREW_SVN]
-else []
-end.each do |key|
-  exec ENV[key], *ARGV if ENV[key] && executable?(ENV[key])
-end
-
-brew_version = File.expand_path("#{D}/../../../bin/#{F}")
-exec brew_version, *ARGV if executable? brew_version
-
-`/usr/bin/which -a #{F} 2>/dev/null`.split("\n").each do |path|
-  exec path, *ARGV unless path == "/usr/bin/#{F}"
-end
-
-popup_stub = false
-if executable? "/usr/bin/xcode-select"
+  fi
+  exec "$@"
+}
+
+F="${0##*/}"
+D="$(quiet_cd "${0%/*}/" && pwd -P)"
```

Comment:
Again: I know it's not your issue but a nicer variable name would be great here.